doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
```

Comment:
I read on sessions in CppCMS before, but they are cookie based only. I've never seen cookie sessions been used for any other API before, so I doubt this is good design..? Of course it would be more comfortable because it is there already, but there must be a reason why APIs use JWTs, oAuth and other mechanism in favour of cookies.

I understand your problem with the dependencies and I'll see what I can do about it, but I'm not promising anything. I was talking about crypto docu by the way, not sessions. I'm using crypto also for other taks (e.g. password hashing). The only docu I could find is the generated doxygen docu that doesn't give much info at all.